azurerm/resource_arm_storage_account.go

```diff
@@ -676,24 +747,43 @@ func resourceArmStorageAccountRead(d *schema.ResourceData, meta interface{}) err
 		if endpoints := props.SecondaryEndpoints; endpoints != nil {
 			if blob := endpoints.Blob; blob != nil {
 				d.Set("secondary_blob_endpoint", blob)
+				if v := endpoints.Blob; v != nil {
+					if u, err := url.Parse(*v); err != nil {
+						d.Set("secondary_blob_host", u.Host)
+					}
+				}
+
 				sscs := fmt.Sprintf("DefaultEndpointsProtocol=https;BlobEndpoint=%s;AccountName=%s;AccountKey=%s",
 					*blob, *resp.Name, *accessKeys[1].Value)
 				d.Set("secondary_blob_connection_string", sscs)
 			} else {
 				d.Set("secondary_blob_endpoint", "")
+				d.Set("secondary_blob_host", "")
 				d.Set("secondary_blob_connection_string", "")
 			}
 
 			if endpoints.Queue != nil {
 				d.Set("secondary_queue_endpoint", endpoints.Queue)
+				if v := endpoints.Queue; v != nil {
+					if u, err := url.Parse(*v); err != nil {
```

Comment:
We should either be checking for an error and bubbling it up or refactor this to not use `url.Parse()`